vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/oceanstor/OceanStor.java

```diff
@@ -0,0 +1,506 @@
+        if (jsonArray != null) {
+            for (int i = 0; i < jsonArray.length(); i++) {
+                JSONObject volume = jsonArray.getJSONObject(i);
+                volumes.add(VolumeMOBuilder.build(volume));
+            }
+        }
+
+        return volumes;
+    }
+
+    public List<VolumeMO> listVolumes(String poolId) throws Exception {
+        List<VolumeMO> volumes = new ArrayList<>();
+
+        JSONArray jsonArray = client.listVolumes(poolId);
+        if (jsonArray != null) {
+            for (int i = 0; i < jsonArray.length(); i++) {
+                JSONObject volume = jsonArray.getJSONObject(i);
+                volumes.add(VolumeMOBuilder.build(volume));
+            }
+        }
+
+        return volumes;
+    }
+
+    public List<StoragePoolMO> listStoragePools() throws Exception {
+        List<StoragePoolMO> pools = new ArrayList<>();
+
+        JSONArray jsonArray = client.listStoragePools();
+        if (jsonArray != null) {
+            for (int i = 0; i < jsonArray.length(); i++) {
+                JSONObject pool = jsonArray.getJSONObject(i);
+                pools.add(StoragePoolMOBuilder.build(pool));
+            }
+        }
+
+        return pools;
+    }
+
+    public StoragePoolMO getStoragePool(String poolId) throws Exception {
+        JSONObject pool = client.getStoragePool(poolId);
+        return StoragePoolMOBuilder.build(pool);
+    }
+
+    private JSONObject getISCSIInitiator(String initiator) throws Exception {
+        JSONObject jsonObject = client.getISCSIInitiator(initiator);
+        if (jsonObject == null) {
+            String msg = String.format("ISCSI initiator %s doesn't exist.", initiator);
+            throw new Exception(msg);
+        }
+
+        if (jsonObject.getInt("RUNNINGSTATUS") != RUNNING_STATUS.ONLINE.getValue()) {
+            String msg = String.format("ISCSI initiator %s isn't online.", initiator);
+            throw new Exception(msg);
+        }
+
+        return jsonObject;
+    }
+
+    private List getFCInitiators(String[] initiators) throws Exception {
+        List iniList = new ArrayList<JSONObject>();
+
+        for (String i: initiators) {
+            JSONObject jsonObject = client.getFCInitiator(i);
+            if (jsonObject == null) {
+                continue;
+            }
+
+            if (jsonObject.getInt("RUNNINGSTATUS") != RUNNING_STATUS.ONLINE.getValue()) {
+               continue;
+            }
+
+            iniList.add(jsonObject);
+        }
+
+        if (iniList.isEmpty()) {
+            String msg = String.format("No any of FC initiators %s exist and online.", initiators.toString());
+            throw new Exception(msg);
+        }
+
+        return iniList;
+    }
+
+    private JSONObject getHost(ConnectMO connect, boolean onlyQuery) throws Exception {
+        JSONObject iscsiInitiator = null;
+        List fcInitiators = null;
+
+        switch (connect.attachProtocol) {
+            case ISCSI:
```

Comment:
Hi，connect.attachProtocol is an enum just as
public enum ATTACH_PROTOCOL {
    ISCSI,
    FC,
    ANY,
}
if the attachProtocol is the default， we will try iscsi and fc. And if they are both failed, it will throw a Expection from this method.